app/cdap/components/WrangleHome/Components/OngoingDataExploration/index.tsx

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { useEffect, useState } from 'react';
+import { Box } from '@material-ui/core/';
+import { generateDataForExplorationCard } from './utils';
+import MyDataPrepApi from 'api/dataprep';
+import { Link } from 'react-router-dom';
+import { getCurrentNamespace } from 'services/NamespaceStore';
+import OngoingDataExplorationCard from '../OngoingDataExplorationCard';
+import { switchMap } from 'rxjs/operators';
+import { forkJoin } from 'rxjs/observable/forkJoin';
+import { IResponseData } from './types';
+
+export default function OngoingDataExploration() {
+  const [ongoingExpDatas, setOngoingExpDatas] = useState([]);
+  const [finalArray, setFinalArray] = useState([]);
+
+  const getOngoingData = () => {
+    // Getting the workspace name, path ,workspaceId and name from MyDataPrepApi.getWorkspaceList API and
+    //  using these in params and requestBody to get Data quality from MyDataPrepApi.execute API
+    MyDataPrepApi.getWorkspaceList({
+      context: 'default',
+    })
+      .pipe(
+        switchMap((res: IResponseData) => {
+          const workspaces = res.values.map((item) => {
+            const params = {
+              context: 'default',
+              workspaceId: item.workspaceId,
+            };
+            const requestBody = {
+              directives: item.directives,
+              limit: 1000,
+              insights: {
+                name: item.sampleSpec.connectionName,
+                workspaceName: item.workspaceName,
+                path: item?.sampleSpec?.path,
+                visualization: {},
+              },
+            };
+
+            setOngoingExpDatas((current) => [
+              ...current,
+              {
+                connectionName:
+                  item?.sampleSpec?.connectionName === undefined
+                    ? 'Upload'
+                    : item?.sampleSpec?.connectionName,
+                workspaceName: item.workspaceName,
+                recipeSteps: item.directives.length,
+                dataQuality: null,
+                workspaceId: item.workspaceId,
+              },
+            ]);
+            return MyDataPrepApi.execute(params, requestBody);
+          });
+          return forkJoin(workspaces);
+        })
+      )
+      .subscribe((response) => {
```

Comment:
Nit: this handler is given multiple responses, so would `responses` be a better variable name?

Reply:
Done